rgo/component.py

```diff
@@ -98,16 +100,9 @@ def make_srpm(self, tmpdir):
             spec_release = spec_release.decode("utf-8")
 
         _spec_path = os.path.join(workdir, "{!s}.spec".format(spec_name))
+
         if self.git:
-            if self.distgit:
-                # we're using downstream dist-git spec as a build recipe,
-                # but we want to use the version from upstream git
-                version, release = self.git.describe(self.name)
-            else:
-                # we're using a spec that is part of the git repo we're building from
-                # if spec and git versions do not match, it is considered an intention
-                # and the spec version prevails
-                version, release = self.git.describe(self.name, spec_version=spec_version)
+            version, release = self.git.describe(self.name, spec_version=spec_version, version_from=self.version_from)
```

Comment:
I changed the default: distgit -> version from git; otherwise -> version from spec
If you specify version_from in the overlay, you get what you ask for